Sources/Core/ObjectiveCDictionaryExtension.swift

```diff
@@ -207,28 +202,28 @@ extension ObjCFileRenderer {
                         "[\(dictionary) setObject:[NSNull null] forKey:@\"\(param)\"];"
                     ]}
             }
-        case .oneOf(types: let avTypes):
+        case .oneOf(types: _):
+            // oneOf (ADT) types have a dictionaryObjectRepresentation method we will use here
             return
                 ObjCIR.ifElseStmt("\(propIVarName) != nil") {[
-                    ObjCIR.switchStmt("\(propIVarName).internalType") {
-                        avTypes.enumerated().map { (_, schema) -> ObjCIR.SwitchCase in
-                            return ObjCIR.caseStmt(self.className+propIVarName.snakeCaseToCamelCase()+"InternalType"+ObjCADTRenderer.objectName(schema)) {[
-                                    "[\(dictionary) setObject:[\(propIVarName) dictionaryObjectRepresentation] forKey:@\"\(param)\"];"
-                                ]}
-                        }
-                    }
+                    "[\(dictionary) setObject:[\(propIVarName) dictionaryObjectRepresentation] forKey:@\"\(param)\"];"
                 ]} {[
                     "[\(dictionary) setObject:[NSNull null] forKey:@\"\(param)\"];"
                 ]}
         case .reference(with: let ref):
             return ref.force().map {
-                renderAddObjectStatement(param, $0, dictionary)
+                renderAddToDictionaryStatement(param, $0, dictionary)
                 } ?? {
                     assert(false, "TODO: Forward optional across methods")
                     return ""
                 }()
-        default:
-            return ""
+        case .map(valueType: .none), .array(.none), .set(.none):
+            return
+                ObjCIR.ifElseStmt("\(propIVarName) != nil") {[
+                    "[\(dictionary) setObject:\(propIVarName) forKey:@\"\(param)\"];"
+                ]} {[
+                    "[\(dictionary) setObject:[NSNull null] forKey:@\"\(param)\"];"
```

Comment:
Do we really want to set `NSNull` in this case? if the prop is not there we may should just not add it to the dictionary at all.

Reply:
I think we want to because this would indicate that the property was actually set to `null` vs being `nil` because it was never set.

Reply:
Yeah I remember we talked about that a while back and the capabilities of our API ...